src/components/TextInput/TextInput.jsx

```diff
@@ -0,0 +1,177 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames/bind';
+import styles from './TextInput.scss';
+const cx = classnames.bind( styles );
+
+const TextInput = ( {
+	autoComplete,
+	ariaDescribedBy,
+	disabled,
+	isMultiline,
+	inputRef,
+	id,
+	label,
+	maxLength,
+	placeholder,
+	onBlur,
+	onChange,
+	onClick,
+	onFocus,
+	onInvalid,
+	pattern,
+	readOnly,
+	required,
+	type,
+} ) => {
+	// Class names used by both <input> and <textarea>.
+	const sharedClassNames = {
```

Comment:
you could use attribute selectors here and avoid cx, fyi

Reply:
Nice! will do